pkg/sdk/procedures_validations_gen.go

```diff
@@ -172,3 +178,128 @@ func (opts *DescribeProcedureOptions) validate() error {
 	}
 	return JoinErrors(errs...)
 }
+
+func (opts *CallProcedureOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if !ValidObjectIdentifier(opts.name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	return JoinErrors(errs...)
+}
+
+func (opts *CreateAndCallForJavaProcedureOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if !valueSet(opts.RuntimeVersion) {
+		errs = append(errs, errNotSet("CreateAndCallForJavaProcedureOptions", "RuntimeVersion"))
+	}
+	if !valueSet(opts.Handler) {
+		errs = append(errs, errNotSet("CreateAndCallForJavaProcedureOptions", "Handler"))
+	}
+	if !ValidObjectIdentifier(opts.ProcedureName) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if !ValidObjectIdentifier(opts.Name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if valueSet(opts.Returns) {
+		if !exactlyOneValueSet(opts.Returns.ResultDataType, opts.Returns.Table) {
+			errs = append(errs, errExactlyOneOf("CreateAndCallForJavaProcedureOptions.Returns", "ResultDataType", "Table"))
+		}
+	}
+	return JoinErrors(errs...)
+}
+
+func (opts *CreateAndCallForScalaProcedureOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if !valueSet(opts.RuntimeVersion) {
+		errs = append(errs, errNotSet("CreateAndCallForScalaProcedureOptions", "RuntimeVersion"))
+	}
+	if !valueSet(opts.Handler) {
+		errs = append(errs, errNotSet("CreateAndCallForScalaProcedureOptions", "Handler"))
+	}
+	if !ValidObjectIdentifier(opts.ProcedureName) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if !ValidObjectIdentifier(opts.Name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if valueSet(opts.Returns) {
+		if !exactlyOneValueSet(opts.Returns.ResultDataType, opts.Returns.Table) {
+			errs = append(errs, errExactlyOneOf("CreateAndCallForScalaProcedureOptions.Returns", "ResultDataType", "Table"))
+		}
+	}
+	return JoinErrors(errs...)
+}
+
+func (opts *CreateAndCallForJavaScriptProcedureOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
+	}
+	var errs []error
+	if !valueSet(opts.ProcedureDefinition) {
+		errs = append(errs, errNotSet("CreateAndCallForJavaScriptProcedureOptions", "ProcedureDefinition"))
+	}
+	if !ValidObjectIdentifier(opts.ProcedureName) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
```

Comment:
For invalid identifiers other than "name" I would suggest using errInvalidIdentifier() function. I know this is generated, but it can only generate it if you specify validation g.ValidIdentifierIfSet (or something like that)